packages/opencrvs/test/Adaptor.test.js

```diff
@@ -0,0 +1,196 @@
+        },
+        entry: birthRecordData,
+      })
+    )(state);
+
+    expect(finalState.data).to.eql({
+      resourceType: 'Bundle',
+      type: 'document',
+      entry: birthRecordData,
+    });
+  });
+});
+
+describe('post', () => {
+  it('makes a post request to create a birth notification', async () => {
+    testServer
+      .intercept({
+        path: '/notification',
+        method: 'POST',
+        headers: {
+          Authorization: 'Bearer fake-token',
+          'Content-type': 'application/json',
+        },
+      })
+
+      .reply(200, {
+        resourceType: 'Bundle',
+        type: 'document',
+        entry: birthRecordData,
+      });
+    state.data = birthRecordData;
+
+    const finalState = await execute(
+      post('/notification', {
+        resourceType: 'Bundle',
+        type: 'document',
+        meta: {
+          lastUpdated: new Date().toISOString(),
+        },
+        entry: birthRecordData,
+      })
+    )(state);
+
+    expect(finalState.data).to.eql({
+      resourceType: 'Bundle',
+      type: 'document',
+      entry: birthRecordData,
+    });
+  });
+});
+
+describe('createBirthRecord', () => {
+  it('successfully creates a birth record', async () => {
+    testServer
+      .intercept({
+        path: '/notification',
+        method: 'POST',
+        headers: {
+          Authorization: 'Bearer fake-token',
+          'Content-type': 'application/json',
+        },
+      })
+
+      .reply(200, {
+        resourceType: 'Bundle',
+        type: 'document',
+        entry: birthRecordData,
+      });
+    state.data = birthRecordData;
+
+    const finalState = await execute(
+      createBirthNotification(state => state.data)
+    )(state);
+
+    expect(finalState.data).to.eql({
+      resourceType: 'Bundle',
+      type: 'document',
+      entry: birthRecordData,
+    });
+  });
+});
+
+describe('queryEvents', () => {
+  it('successfully searches for an event', async () => {
+    testServer
+      .intercept({
+        path: '/graphql',
+        method: 'POST',
+        headers: {
+          Authorization: 'Bearer fake-token',
+          'Content-type': 'application/json',
+        },
+      })
+
+      .reply(200, {
+        data: {
+          searchEvents: {
+            totalItems: 0,
+            results: [],
+            __typename: 'EventSearchResultSet',
+          },
+        },
+      });
+
+    const finalState = await execute(
+      queryEvents({
+        event: 'birth',
+        registrationStatuses: ['REGISTERED'],
+        childGender: 'male',
+        dateOfRegistrationEnd: '2022-12-31T23:59:59.999Z',
+        dateOfRegistrationStart: '2021-11-01T00:00:00.000Z',
+        declarationJurisdictionId: '',
+        eventLocationId: '704b9706-d729-4834-8656-05b562065deb',
+        fatherFirstNames: 'Dad',
+        motherFirstNames: 'Mom',
+      })
+    )(state);
+
+    expect(finalState.data).to.eql({
+      data: {
+        searchEvents: {
```

Comment:
The nice thing about this test is that it's shown me how the function works without having to run the code. Which is great because I can form opinions on it :)

Do we really want to return the raw search object? Why do I have to do `state.data.searchEvents.results` in my job code?

I think you should just write the results array to `state.data`. Much easier for users.